dataprofiler/data_readers/base_data.py

```diff
@@ -93,7 +99,12 @@ def data_format(self, value):
     @property
     def file_encoding(self):
         if not self._file_encoding:
-            return sys.getdefaultencoding()
+            # set to default, detect if not StringIO
+            self._file_encoding = sys.getdefaultencoding()
```

Comment:
Shouldn't we detect the file encoding first?

Also, can this error? I'm not sure `sys.getdefaultcoding()` is correct here

https://stackoverflow.com/questions/15530635/why-is-sys-getdefaultencoding-different-from-sys-stdout-encoding-and-how-does

It appears to be discouraged 

https://stackoverflow.com/questions/3828723/why-should-we-not-use-sys-setdefaultencodingutf-8-in-a-py-script

Reply:
updated to use locale and fallback on sys.getfilesystemencoding